misc/scripts/bwrap.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+#     ____                  __        ____
+#    / __ \____ ___________/ /_____ _/ / /
+#   / /_/ / __ `/ ___/ ___/ __/ __ `/ / /
+#  / ____/ /_/ / /__(__  ) /_/ /_/ / / /
+# /_/    \__,_/\___/____/\__/\__,_/_/_/
+#
+# Copyright (C) 2020-present
+#
+# This file is part of Pacstall
+#
+# Pacstall is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, version 3 of the License
+#
+# Pacstall is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Pacstall. If not, see <https://www.gnu.org/licenses/>.
+
+function safe_source() {
+    local input="${1}"
+    mkdir /tmp/pacstall 2>/dev/null
+    safeenv="$(sudo mktemp -p "${PACDIR}")"
+    sudo chmod +r "$safeenv"
+    bwrapenv="$(sudo mktemp -p "${PACDIR}" -t "bwrapenv.XXXXXXXXXX")"
+    sudo chmod +r "$bwrapenv"
+    export bwrapenv
+    export safeenv
+
+    tmpfile="$(sudo mktemp -p "${PACDIR}")"
+    local allsource allsums src sum a_sum known_hashsums_src=("b2" "sha512" "sha384" "sha256" "sha224" "sha1" "md5") known_archs_src=("amd64" "arm64" "armel" "armhf" "i386" "mips64el" "ppc64el" "riscv64" "s390x")
```

Comment:
`known_archs_src` should come from, you guessed it, `dpkg-architecture --list-known`.